lib/base-compiler.ts

```diff
@@ -1254,7 +1254,7 @@ export class BaseCompiler implements ICompiler {
         }
     }
 
-    getExecutableFilename(dirPath, outputFilebase, key?) {
+    getExecutableFilename(dirPath: string, outputFilebase: string, key?) {
```

Comment:
do we know what the `key?` type is? looks like `: any` from `getOutputFilename` though I suspect it's possible to imrpove on that...though not in this PR necessarily

Reply:
I think key "should" be something that looks like the result of https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/base-compiler.ts#L1681 and in some cases what's returned by https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/base-compiler.ts#L1685 - in theory at least

Reply:
I'll probably defer this to another PR, this PR is a big enough review already :)

Reply:
100%